### PR TITLE
Feat added POST superadmin endpoint

### PR DESCRIPTION
**Pull request by marveldo**
> > ## Description
> > This pull request introduces an endpoint for the registeration of admin users
> > The changes include:
> > 
> > * **Endpoint Addition:** Added `POST /api/v1/superadmin/register` for creating super admins.
> > * **Error Handling:** Implemented checks for invalid data and already existing accounts
> > * **Testing:** Added pytest cases to ensure functionality and edge cases are covered.
> 
> > ## Motivation and Context
> > This feature is essential to allow Admins to be registered
> > ## How Has This Been Tested?
> > 
> > * **Unit Tests:** Added tests to validate the End point responsible to registering superusers
> 
> > Tests include:
> > 
> > * Adding a superuser
> > * Checking whether the is_super_admin field of the user created reads true
> > * Handling validation errors and creation of multiple accounts
> > 
> > ## Screenshots:
> > ### Successful Request
> > ##Screenshot of Test
Screenshot 2024-07-24 214513

)
> 
> > ## Types of changes
> > 
> > * []   Bug fix (non-breaking change which fixes an issue)
> > * [x]    New feature (non-breaking change which adds functionality)
> > * [ ]    Breaking change (fix or feature that would cause existing functionality to change)
> > 
> > ## Checklist:
> > 
> > * [x]    My code follows the code style of this project.
> > * []   My change requires a change to the documentation.
> > * []   I have updated the documentation accordingly.
> > * [x]    I have read the **CONTRIBUTING** document.
> > * [x]    I have added tests to cover my changes.
> > * [x]    All new and existing tests passed.

​

### CODE REVIEW

File: api/v1/routes/superadmin.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        message= 'User Created Successfully',
+        data={
+            'id' : user_created.id ,
+            'first_name' : user_created.first_name,
+            'last_name': user_created.last_name,
+            'username' : user_created.username,
+             'email': user_created.email,
+            'created_at' : str(user_created.created_at)
+        }
+    )
```
Comment (joboy-dev) [inline api/v1/routes/superadmin.py:38]: Change that `data = {...}` to `user_created.to_dict()`

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Attend to requests

File: api/v1/routes/superadmin.py (hunk 1 above)
Comment (marveldo) [reply api/v1/routes/superadmin.py:38]: user_created to dict is not working @joboy-dev  and im using 3.10
